File: AfterMarketParts/src/test/java/javaApiScript/ApiScriptOverJavaTest.java
package javaApiScript;

import static org.junit.Assert.*;

import java.util.List;

import org.junit.Test;


import javaApiScriptOverJava.ApiOperationOneOverJava;
import javaApiScriptOverJava.ApiOperationOverJava;
import javaApiScriptOverJava.ApiOperationTwoOverJava;
import javaApiScriptOverJava.ApiWrapperOverJava;
import api.ApiStatus;
import api.OperationResult;

public class ApiScriptOverJavaTest {

	@Test
	public void itShouldAllowYouToScriptASingleApiCall() {
		ApiOperationOverJava apiOperation = new ApiOperationOneOverJava("test");
		OperationResult result = ApiWrapperOverJava.call(apiOperation);
		verifyResult(result, ApiStatus.OK, "operationOne performed on test");
	}

	@Test
	public void itShouldAllowYouToPassMultipleOperationsToTheApi() {
		ApiOperationOverJava apiOperation1 = new ApiOperationOneOverJava("test");
		ApiOperationOverJava apiOperation2 = new ApiOperationTwoOverJava("testing", 123);
		List<OperationResult> results = ApiWrapperOverJava.call(apiOperation1, apiOperation2);
		assertEquals(results.size(), 2);
		verifyResult(results.get(0), ApiStatus.OK, "operationOne performed on test");
		verifyResult(results.get(1), ApiStatus.OK, "operationTwo performed on testing 123");
	}

	private void verifyResult(OperationResult result, ApiStatus expectedStatus, String expectedMessage) {
		assertEquals(result.getStatus(), expectedStatus);
		assertEquals(result.getMessage(), expectedMessage);
	}
}

File: fodder/src/test/scala/scalaApiScript/ApiScriptOverJavaTest.scala
package scalaApiScript
import org.scalatest._

import scalaApiScriptOverJava.ApiWrapper;
import javaSupport.api.ApiStatus._
import javaSupport.api.OriginalApi
import javaSupport.api.OperationResult

class ApiScriptOverJavaTest extends FunSpec with ShouldMatchers {
  
  it("should allow you to pass script a single API call") {
	  val result = ApiWrapper.call(_.operationOne("test"))
	  result.head.getStatus should be(OK)
	  result.head.getMessage should be("operationOne performed on test")
  }
  
  it("should allow you to pass multiple operations to the API") {
      val operations = Seq[OriginalApi => OperationResult](
          _.operationOne("test"),
          _.operationTwo("testing", 123)
      )
      
	  val results = ApiWrapper.call(operations: _*)
	  
	  results map (_.getStatus()) should be (Seq(OK, OK))
	  results map (_.getMessage()) should be (Seq(
	      "operationOne performed on test", 
	      "operationTwo performed on testing 123"))
  }
}
